### ISSUE TITLE
--reload not working using systemd service in versions 0.14.x

### ISSUE THREAD

**Issue by teemu77** [pgjones/hypercorn#69]
Latest working version was hypercorn==0.13.2. Tested both 0.14.0 and 0.14.1. OS is Ubuntu 20.04.4 LTS.

My service config having working --reload with version 0.13.2:
/etc/systemd/system/hypercorn-myservice_v1@dev.service
```
[Unit]
Description=%N daemon
After=network.target
StartLimitIntervalSec=0

[Service]
Type=simple
RuntimeDirectory=%i/%p
User=myuser
Group=mygroup
UMask=117
WorkingDirectory=/srv/fastapi/%i/myservice
ExecStart=/srv/fastapi/%i/myservice/venv/bin/hypercorn \
          --pid %t/%i/%p/pid \
          --bind unix:%t/%i/%p/socket \
          --workers 1 \
          --root-path "/myservice/v1" \
          --reload \
          app.v1.main:app
ExecReload=/bin/kill -s HUP $MAINPID
ExecStop=/bin/kill -s TERM $MAINPID
PrivateTmp=true
Restart=always

[Install]
WantedBy=multi-user.target
```

**Comment by ll2pakll** [comment]
My reload doesn't work either. After changing the code, when you try to access the page through the browser, the page becomes inaccessible and the server shuts down. Here is my code:

> from hypercorn.config import Config
> import trio
> from hypercorn.trio import serve
> from starlette.applications import Starlette
> from starlette.responses import JSONResponse
> from starlette.routing import Route
> 
> config = Config()
> config.use_reloader = True
> config.bind = ["localhost:12345"]  # As an example configuration setting
> 
> 
> async def homepage(request):
>     return JSONResponse({'hello': 'love'})
> 
> 
> app = Starlette(debug=True, routes=[
>     Route('/', homepage),
> ])
> 
> trio.run(serve, app, config)

**Comment by pgjones** [comment]
@teemu77 are you also using the `serve` api? If so you will need to manage reloading directly as it is no longer supported via the serve API. This is due to it being unreliable, and requiring worker management to fix.

I am surprised it becomes inaccessible and the server shuts down though, any crash output?

**Comment by ll2pakll** [comment]
> from hypercorn.config import Config import trio from hypercorn.trio import serve from starlette.applications import Starlette from starlette.responses import JSONResponse from starlette.routing import Route
> 
> config = Config() config.use_reloader = True config.bind = ["localhost:12345"] # As an example configuration setting
> 
> async def homepage(request): return JSONResponse({'hello': 'love'})
> 
> app = Starlette(debug=True, routes=[ Route('/', homepage), ])
> 
> trio.run(serve, app, config)

No, the program just stops with "Process finished with exit code 0"

**Comment by teemu77** [comment]
I had no issues with crashing. All is working except the reloading.

Not really sure what `serve` API is so I'm most probably not using it =)

I see that my orignal issue is pretty laborious to replicate, but did I some more tests with a really basic setup. The problem actually is not related to systemd after all.

First I upgraded hypercorn-0.13.2 -> hypercorn-0.14.1
```
(venv)$ python -V
Python 3.8.10
(venv)$ pip uninstall hypercorn
Found existing installation: hypercorn 0.13.2
Uninstalling hypercorn-0.13.2:
  Would remove:
    /srv/fastapi/dev/myservice/venv/bin/hypercorn
    /srv/fastapi/dev/myservice/venv/lib/python3.8/site-packages/hypercorn-0.13.2.dist-info/*
    /srv/fastapi/dev/myservice/venv/lib/python3.8/site-packages/hypercorn/*
Proceed (Y/n)? y
  Successfully uninstalled hypercorn-0.13.2
(venv)$ pip install hypercorn
Collecting hypercorn
  Using cached Hypercorn-0.14.1-py3-none-any.whl (57 kB)
Requirement already satisfied: wsproto>=0.14.0 in /srv/fastapi/dev/myservice/venv/lib/python3.8/site-packages (from hypercorn) (1.2.0)
Requirement already satisfied: priority in /srv/fastapi/dev/myservice/venv/lib/python3.8/site-packages (from hypercorn) (2.0.0)
Requirement already satisfied: toml in /srv/fastapi/dev/myservice/venv/lib/python3.8/site-packages (from hypercorn) (0.10.2)
Requirement already satisfied: h2>=3.1.0 in /srv/fastapi/dev/myservice/venv/lib/python3.8/site-packages (from hypercorn) (4.1.0)
Requirement already satisfied: h11 in /srv/fastapi/dev/myservice/venv/lib/python3.8/site-packages (from hypercorn) (0.13.0)
Requirement already satisfied: hyperframe<7,>=6.0 in /srv/fastapi/dev/myservice/venv/lib/python3.8/site-packages (from h2>=3.1.0->hypercorn) (6.0.1)
Requirement already satisfied: hpack<5,>=4.0 in /srv/fastapi/dev/myservice/venv/lib/python3.8/site-packages (from h2>=3.1.0->hypercorn) (4.0.0)
Installing collected packages: hypercorn
Successfully installed hypercorn-0.14.1
```
Some minimal FastAPI code:
```
$ cat app/v1/main.py
from fastapi import FastAPI
app = FastAPI()
@app.get("/")
async def read_root():
    return {"Hello": "World"}
```
Start Hypercorn and leave running in background. This uses unix-socket just like my service.
```
(venv)$ hypercorn --pid ./pid --bind unix:./socket --workers 1 --root-path /myservice/v1 --reload app.v1.main:app &
```
Socket is working:
```
$ curl --unix-socket ./socket http
{"Hello":"World"}
```
Modify main.py:
```
$ sed -i 's/World/Universe/' app/v1/main.py
```
This is interesting, Hypercorn seems to be somewhat aware of the modification since the background process echoes:
```
(venv)$ [2022-09-02 13:36:45 +0300] [1772702] [INFO] Running on http://unix:./socket (CTRL + C to quit)
```
But the file is not reloaded:
```
(venv)$ curl --unix-socket ./socket http
{"Hello":"World"}
```
Did a test without sockets and results are identical.
```
(venv)$ hypercorn --workers 1 --root-path /myservice/v1 --reload app.v1.main:app &
```
```
$ curl http://127.0.0.1:8000
{"Hello":"Universe"}
```
Modify main.py:
```
$ sed -i 's/Universe/EnterTheVoid/' app/v1/main.py
```
Once again background process immediately echoes:
```
[2022-09-02 13:55:17 +0300] [1772954] [INFO] Running on http://127.0.0.1:8000 (CTRL + C to quit)
```
But still the code change isn't picked up by Hypercorn.
```
$ curl http://127.0.0.1:8000
{"Hello":"Universe"}
```

**Comment by pgjones** [comment]
Should be fixed in a5bf3caac600dedf2350bdd62f2e5ec92bc6dbc0
